commons-numbers-complex/src/test/java/org/apache/commons/numbers/complex/CStandardTest.java

```diff
@@ -865,12 +884,29 @@ void testImplicitTrig() {
             final double re = next(rng);
             final double im = next(rng);
             final Complex z = complex(re, im);
-            final Complex iz = z.multiplyImaginary(1);
-            assertComplex(z.asin(), iz.asinh().multiplyImaginary(-1));
-            assertComplex(z.atan(), iz.atanh().multiplyImaginary(-1));
-            assertComplex(z.cos(), iz.cosh());
-            assertComplex(z.sin(), iz.sinh().multiplyImaginary(-1));
-            assertComplex(z.tan(), iz.tanh().multiplyImaginary(-1));
+            final Complex iz1 = z.multiplyImaginary(1);
```

Comment:
The code repetition here is prone to error. The operations should be tested using TestUtils.assert with two operations created using composition:
```Java
ComplexUnaryOperator<ComplexNumber> mi1 = (x, y, out) -> ComplexFunctions.multiplyImaginary(x, y, 1, out);
ComplexUnaryOperator<ComplexNumber> mim1 = (x, y, out) -> ComplexFunctions.multiplyImaginary(x, y, -1, out);

UnaryOperator<Complex> asin1 = c -> c.multiplyImaginary(1).asinh().multiplyImaginary(-1);
ComplexUnaryOperator<ComplexNumber> asin2 = mi1.andThen(ComplexFunctions::asinh).andThen(mim1);

// Test the two functions produce the same result
// Test it is the expected result
```